presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -769,25 +738,103 @@ public void revokeTablePrivileges(String databaseName, String tableName, String
         }
     }
 
+    // GRANT and REVOKE prioritize roles over user names, given a string which is both, a role name and a user.
```

Comment:
I would expect that GRANT/REVOKE would prioritize user names over roles, but you should check the spec and how other databases behave.